activerecord/lib/arel/insert_manager.rb

```diff
@@ -33,13 +33,13 @@ def insert(fields)
           @ast.columns << column
           values << value
         end
-        @ast.values = create_values values, @ast.columns
+        @ast.values = create_values(values)
       end
       self
     end
 
-    def create_values(values, columns)
-      Nodes::Values.new values, columns
+    def create_values(values)
+      Nodes::ValuesList.new([values])
```

Comment:
Is the extra array allocation here anything to be concerned about?

Reply:
Actually this is called in the `_insert_record`, so it would allocate the extra array per one insert.

https://github.com/rails/rails/blob/7971fc4b49979a8d1ded09e311c5c09a09af5b93/activerecord/lib/active_record/persistence.rb#L382-L389

I've benchmarked with https://gist.github.com/kamipo/8346d67309405ef5f1e38e085df15681, but I can't find any difference.

The `_insert_record` is already much improved enough in 5.2 (bbae710a405ce92074c1666dcf859196c29e2ed2), so the one extra array allocation is acceptable to me.

Reply:
Great!